partiql-value/src/lib.rs

```diff
@@ -887,6 +892,21 @@ impl List {
     pub fn iter(&self) -> ListIter {
         ListIter(self.0.iter())
     }
+
+    #[inline]
+    pub fn reserve(&mut self, additional: usize) {
```

Comment:
Just curious, is `reserve` used for some performance improvements?

Reply:
The usage in `extend` is idiomatic. Yes, it can be more performant to perform any allocations/moves prior to inserting a sequence of values rather than in the midst.